src/components/program-settings.js

```diff
@@ -139,94 +139,64 @@ class ProgramSettings extends React.Component {
         },
     }
 
-    handleChange = e => {
-        e.preventDefault()
-        const valueInput = e.target.value
-        switch (e.target.name) {
+    chooseSetting = (name, value) => {
+        switch (name) {
             case 'teiDownload':
-                e.target.value > maxValues.teiDownload
-                    ? (e.target.value = maxValues.teiDownload)
-                    : (e.target.value = valueInput)
+                value > maxValues.teiDownload
+                    ? (value = parseInt(maxValues.teiDownload))
+                    : (value = parseInt(value))
                 break
             case 'teiDBTrimming':
-                e.target.value > maxValues.teiDBTrimming
-                    ? (e.target.value = maxValues.teiDBTrimming)
-                    : (e.target.value = valueInput)
+                value > maxValues.teiDBTrimming
+                    ? (value = parseInt(maxValues.teiDBTrimming))
+                    : (value = parseInt(value))
                 break
             case 'teReservedDownload':
-                e.target.value > maxValues.teReservedDownload
-                    ? (e.target.value = maxValues.teReservedDownload)
-                    : (e.target.value = valueInput)
+                value > maxValues.teReservedDownload
+                    ? (value = parseInt(maxValues.teReservedDownload))
+                    : (value = parseInt(value))
                 break
             case 'teReservedDBTrimming':
-                e.target.value > maxValues.teReservedDBTrimming
-                    ? (e.target.value = maxValues.teReservedDBTrimming)
-                    : (e.target.value = valueInput)
+                value > maxValues.teReservedDBTrimming
+                    ? (value = parseInt(maxValues.teReservedDBTrimming))
+                    : (value = parseInt(value))
                 break
             case 'eventsDownload':
-                e.target.value > maxValues.eventsDownload
-                    ? (e.target.value = maxValues.eventsDownload)
-                    : (e.target.value = valueInput)
+                value > maxValues.eventsDownload
+                    ? (value = parseInt(maxValues.eventsDownload))
+                    : (value = parseInt(value))
                 break
             case 'eventsDBTrimming':
-                e.target.value > maxValues.eventsDBTrimming
-                    ? (e.target.value = maxValues.eventsDBTrimming)
-                    : (e.target.value = valueInput)
+                value > maxValues.eventsDBTrimming
+                    ? (value = parseInt(maxValues.eventsDBTrimming))
+                    : (value = parseInt(value))
```

Comment:
All of these would be MUCH simpler written as `value = Math.min(maxValues.eventDBTrimming, parseInt(value))`, as I mentioned in another review.  The `parseInt` on `maxValues.x` isn't necessary, since they should already be numbers, and this way the assignment only happens once.  Would you mind rewriting each of these cases that way?

Reply:
Sure I rewrite every case like this:
`value = Math.min(maxValues.teiDownload, parseInt(value))`